packages/chat-stateful-client/src/ChatContext.ts

```diff
@@ -357,6 +357,36 @@ export class ChatContext {
     }
   }
 
+  /**
+   * Tees any errors encountered in an function to the state.
+   *
+   * If the function succeeds, clears associated errors from the state.
+   *
+   * @param f Function to execute.
+   * @param target The error target to tee error to.
+   * @param clearTargets The error targets to clear errors for if the function succeeds. By default, clears errors for `target`.
+   * @returns Result of calling `f`. Also re-raises any exceptions thrown from `f`.
+   * @throws ChatError. Exceptions thrown from `f` are tagged with the failed `target.
+   */
+  public withErrorTeedToState<T>(f: () => T, target: ChatErrorTargets, clearTargets?: ChatErrorTargets[]): () => T {
```

Comment:
* The implementation here is very similar to `asyncTeeErrorToState`. I couldn't figure out a sane way to dedupe. Ideas welcome.
* Unlike `asyncTeeErrorToState`, it was more ergonomic to write this as a higher order function. In a future PR, I'll convert `asyncTeeErrorToState` to also be a higher order function, to make the two consistent.